sdk/core/azure-core/azure/core/configuration.py

```diff
@@ -26,6 +26,58 @@
 
 
 class Configuration(object):
+    """Provides the home for all of the configurable policies in the pipeline.
+
+    A new Configuration object provides no default policies and does not specify in what
+    order the policies will be added to the pipeline. The SDK developer must specify each
+    of the policy defaults as required by the service and use the policies in the
+    Configuration to construct the pipeline correctly, as well as inserting any
+    unexposed/non-configurable policies.
+
+    .. code-block:: python
+
+        class FooServiceClient():
+
+            @staticmethod
+            def create_config(**kwargs):
+                # Here the SDK developer would define the default
+                # config to interact with the service
+                config = Configuration(**kwargs)
+                config.headers_policy = HeadersPolicy({"CustomHeader": "Value"}, **kwargs)
+                config.user_agent_policy = UserAgentPolicy("ServiceUserAgentValue", **kwargs)
+                config.retry_policy = RetryPolicy(**kwargs)
+                config.redirect_policy = RedirectPolicy(**kwargs)
+                config.logging_policy = NetworkTraceLoggingPolicy(**kwargs)
+                config.proxy_policy = ProxyPolicy(**kwargs)
+                config.transport = kwargs.get('transport', RequestsTransport)
+
+            def __init__(self, endpoint=None, credentials=None, configuration=None, **kwargs):
+                config = configuration or FooServiceClient.create_config(**kwargs)
+                transport = config.get_transport(**kwargs)
+                policies = [
+                    config.user_agent_policy,
+                    config.headers_policy,
+                    credentials,
+                    ContentDecodePolicy(),
+                    config.redirect_policy,
+                    config.retry_policy,
+                    config.logging_policy,
+                ]
+                self._pipeline = Pipeline(transport, policies=policies)
+
+    :param connection: Provides the configuration parameters for the transport.
+    :type connection: ~azure.core.ConnectionConfiguration
+    :param headers_policy: Provides parameters for custom or additional headers to be sent with the request.
+    :param proxy_policy: Provides configuration parameters for proxy.
+    :param redirect_policy: Provides configuration parameters for redirects.
+    :param retry_policy: Provides configuration parameters for retries in the pipeline.
+    :param custom_hook_policy: Provides configuration parameters for a custom hook.
+    :param logging_policy: Provides configuration parameters for logging.
+    :param user_agent_policy: Provides configuration parameters to append custom values to the
+     User-Agent header.
+    :param transport: The Http Transport type. E.g. RequestsTransport, AsyncioRequestsTransport,
+     TrioRequestsTransport, AioHttpTransport.
```

Comment:
A new `authentication_policy` was added in https://github.com/Azure/azure-sdk-for-python/pull/5498/files